### PR TITLE
Add Identity OP

### PR DESCRIPTION
**Pull request by shiyutang**
### PR types

New features

### PR changes

OPs

### Describe
Add Identity op which have the same functionality as torch.nn.Identity, if we define the "m" as the identity layer, it creates following results.
Ineut-tensor - poddle to-tensorCmrpy rondon rondnC1z, 20))

### CODE REVIEW

File: python/paddle/fluid/tests/unittests/test_identity_op.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+# limitations under the License.
+
+import unittest
+import numpy as np
+import paddle.fluid as fluid
+import paddle.fluid.core as core
+import paddle
+
+
+class TestDiagonalAPI(unittest.TestCase):
```
Comment (ZHUI) [inline python/paddle/fluid/tests/unittests/test_identity_op.py]: TestDiagonalAPI?

File: python/paddle/nn/layer/common.py [hunk 2]
```diff
@@ -30,6 +30,46 @@ def _npairs(x, n):
...
+        - input: Multi-dimentional tensor with shape :math:`[batch\_size, n1, n2, ...]` .
+        - output: Multi-dimentional tensor with shape :math:`[batch\_size, n1, n2, ...]` .
+
+    Examples:
+        .. code-block:: python
+
+         import numpy
+         import paddle
+         layer = Identity(54)
+         input_tensor = paddle.to_tensor(numpy.random.randn(128, 20))
```
Comment (ZHUI) [inline python/paddle/nn/layer/common.py]: Use paddle.randn, you can delete numpy

File: python/paddle/nn/layer/common.py [hunk 3]
```diff
@@ -30,6 +30,46 @@ def _npairs(x, n):
...
+         input_tensor = input_tensor+1
+         input_tensor.stop_gradient=False
+         input_tensor.register_hook(lambda grad: print('input grad', grad))
+         print('input_tensor.grad', input_tensor.grad)
```
Comment (ZHUI) [inline python/paddle/nn/layer/common.py]: Please show the usage of Identity layer.

**Review by ZHUI — CHANGES_REQUESTED** [review]

File: python/paddle/fluid/tests/unittests/test_identity_op.py (hunk 1 above)
Comment (shiyutang) [reply python/paddle/fluid/tests/unittests/test_identity_op.py]: changed to TestIdentityAPI

File: python/paddle/nn/layer/common.py [hunk 4]
```diff
@@ -30,6 +30,52 @@ def _npairs(x, n):
...
+    Shape:
+        - input: Multi-dimentional tensor with shape :math:`[batch\_size, n1, n2, ...]` .
+        - output: Multi-dimentional tensor with shape :math:`[batch\_size, n1, n2, ...]` .
+
+    Examples:
+        .. code-block:: python
+
+          import paddle
+          layer = paddle.nn.Identity(54)
+          input_tensor = paddle.to_tensor(paddle.randn(shape=[3, 2]))
```
Comment (ZHUI) [inline python/paddle/nn/layer/common.py]: No need for `paddle.to_tensor`

File: python/paddle/nn/layer/common.py [hunk 5]
```diff
@@ -30,6 +30,52 @@ def _npairs(x, n):
...
+
+    Shape:
+        - input: Multi-dimentional tensor with shape :math:`[batch\_size, n1, n2, ...]` .
+        - output: Multi-dimentional tensor with shape :math:`[batch\_size, n1, n2, ...]` .
+
+    Examples:
+        .. code-block:: python
+
+          import paddle
+          layer = paddle.nn.Identity(54)
```
Comment (ZHUI) [inline python/paddle/nn/layer/common.py]: what's the means of `54`?

File: python/paddle/nn/layer/common.py [hunk 6]
```diff
@@ -30,6 +30,52 @@ def _npairs(x, n):
...
+          input_tensor = paddle.to_tensor(paddle.randn(shape=[3, 2]))
+          input_tensor.stop_gradient=False
+          input_tensor = input_tensor+1
+          input_tensor.register_hook(lambda grad: print('input grad', grad))
+          print('input_tensor.grad', input_tensor.grad)
+          out = m(input_tensor)
```
Comment (ZHUI) [inline python/paddle/nn/layer/common.py]: `out = m(input_tensor)` -> `out = layer(input_tensor)`
You can delete the calculation of gradient in the example.

**Review by ZHUI — CHANGES_REQUESTED** [review]

File: python/paddle/nn/layer/common.py (hunk 4 above)
Comment (shiyutang) [reply python/paddle/nn/layer/common.py]: ok

File: python/paddle/nn/layer/common.py (hunk 5 above)
Comment (shiyutang) [reply python/paddle/nn/layer/common.py]: means any parameters, I will delete it

**Review by michaelowenliu — DISMISSED** [review]
LGTM

**Review by michaelowenliu — APPROVED** [review]

**Review by ZHUI — APPROVED** [review]
LGTM

File: python/paddle/nn/layer/common.py [hunk 7]
```diff
@@ -30,6 +30,49 @@ def _npairs(x, n):
     return x
 
 
+class Identity(Layer):
+    r"""
+
+    A placeholder identity operator that is argument-insensitive. For each input :math:`X` ,
```
Comment (jzhang533) [inline python/paddle/nn/layer/common.py:36]: 只支持一个输入，一个输出吧。
这里用`for each`，感觉是支持多个输入的。
